lib/uffizzi/cli/connect.rb

```diff
@@ -45,6 +45,36 @@ def docker_hub
       end
     end
 
+    desc 'docker-registry', 'Connect to any registry implementing the Docker Registry HTTP API protocol'
+    method_option :skip_raise_existence_error, type: :boolean, default: false,
+                                               desc: 'Skip raising an error within check the credential'
+    method_option :update_credential_if_exists, type: :boolean, default: false
+    def docker_registry
+      type = Uffizzi.configuration.credential_types[:docker_registry]
+      credential_exists = credential_exists?(type)
+      handle_existing_credential_options('docker-registry') if credential_exists
+
+      registry_url = ENV['DOCKER_REGISTRY_URL'] || Uffizzi.ui.ask('Registry Domain:')
+      username = ENV['DOCKER_REGISTRY_USERNAME'] || Uffizzi.ui.ask('Username:')
+      password = ENV['DOCKER_REGISTRY_PASSWORD'] || Uffizzi.ui.ask('Password:', echo: false)
+
+      params = {
+        registry_url: prepare_registry_url(registry_url),
+        username: username,
+        password: password,
+        type: type,
+      }
+
+      server = ConfigFile.read_option(:server)
+      response = create_or_update_credential(server, params, create: !credential_exists)
```

Comment:
Instead of passing flag let's sort the logic right here

Reply:
Do you mean extracting the body of the function and copying it in place of the function call? I tried to avoid this.

Reply:
Good abstraction will let you not create 6 copies. And let's avoid using ternary operators.